sdk/go/docker/docker.go

```diff
@@ -0,0 +1,558 @@
+// Copyright 2016-2020, Pulumi Corporation.
+
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"os/exec"
+	"strings"
+
+	"github.com/Masterminds/semver"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/pkg/util/logging"
+	"github.com/pulumi/pulumi/sdk/go/pulumi"
+)
+
+func buildAndPushImageAsync(ctx *pulumi.Context, imageName string, build DockerBuild, repositoryURL string,
```

Comment:
It's fine to keep this function for alignment with other SDKs - but note that it's really not needed here as you have taken care of unwrapping the `Input`s in the calling layer - so the role this function plays in other SDKs is not needed here (and in principle this function should have exactly the same signature as `buildAndPushImageWorkerAsync`).